pkg/controller/generic/qtransform/qtransform.go

```diff
@@ -237,9 +247,17 @@ func (ctrl *QController[Input, Output]) reconcileRunning(ctx context.Context, lo
 			return nil
 		}
 
+		if requeueError != nil {
```

Comment:
I think if `err != nil` here and it's not unwrapped `requeueError`, we should return it, not `requeueError`.

What I mean is: if `Modify` fails e.g. because the resource can't be saved, we should fail hard, not `requeueError`.

Reply:
Hm, don't we already do that?

Above, if `transformError` is not a `RequeueError`, we return it as-is (line 239).

And here, we are in the error case already (because line 240).
If this was indeed a `requeueError`, this line 250 will evaluate to true, and we will return it.

If not, it will return the error as-is on line 254, as you suggested.

Or am I missing something?

Reply:
I mean the following case (for example):

1. `requeueError` - is a wrapped with `nil` `Err()`
2. so at the line 236 we return `nil`
3. `Modify` fails for its own reasons, so `err != nil`
4. on the line 251 we will return `requeueError` masking the real `err`

Reply:
Ok, I think I got it now - please have another look